internal/remoteproc/remoteproc.go

```diff
@@ -17,10 +18,67 @@ const (
 	rprocFirmwareFileName     = "firmware"
 )
 
-var (
-	rprocFirmwareStorePath = rootpath.Join("lib", "firmware")
-	rprocClassPath         = rootpath.Join("sys", "class", "remoteproc")
-)
+var rprocClassPath = rootpath.Join("sys", "class", "remoteproc")
+
+func getCustomFirmwarePath() (string, error) {
+	// Check if kernel has custom firmware path configured
+	customPath, err := os.ReadFile("/sys/module/firmware_class/parameters/path")
+	if err == nil {
+		if path := strings.TrimSpace(string(customPath)); path != "" {
+			return path, nil
+		} else {
+			return "", fmt.Errorf("custom firmware path is empty")
+		}
+	} else {
+		return "", fmt.Errorf("failed to read custom firmware path /sys/module/firmware_class/parameters/path: %w", err)
+	}
```

Comment:
IMHO, error is the wrong semantics here - this not being specified is not an error. This feels equivalent to https://pkg.go.dev/os#LookupEnv which uses `(value, ok)` return values.

I'm also not sure we should be checking for empty here.

```suggestion
func lookupCustomFirmwarePath() (string, bool) {
	// Check if kernel has custom firmware path configured
	customPath, err := os.ReadFile("/sys/module/firmware_class/parameters/path")
	if err != nil {
		return "", false
	}

	return path := strings.TrimSpace(string(customPath))
```